internal/repository/repository_test.go

```diff
@@ -0,0 +1,370 @@
+package repository
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/VTerenya/employees/internal"
+	errs "github.com/VTerenya/employees/internal/errors"
+	"github.com/google/uuid"
+	"github.com/shopspring/decimal"
+)
+
+var (
+	data        *Database   //nolint: gochecknoglobals
+	repos       *Repository //nolint: gochecknoglobals
+	positionIDs []string    //nolint: gochecknoglobals
+	employeeIDs []string    //nolint: gochecknoglobals
+)
+
+func initData() {
+	data = NewDataBase()
+	repos = NewRepo(data)
+	positionIDs = make([]string, 0)
+	employeeIDs = make([]string, 0)
+}
+
+func createPosID() uuid.UUID {
+	id := uuid.New()
+	positionIDs = append(positionIDs, id.String())
+	return id
+}
+
+func createEmpID() uuid.UUID {
+	id := uuid.New()
+	employeeIDs = append(employeeIDs, id.String())
+	return id
+}
+
+func TestGetPositions(t *testing.T) {
+	initData()
+	p := internal.Position{ID: createPosID(), Name: "worker", Salary: decimal.New(500, 0)}
+	repos.AddPosition(&p)
+	testTable := []struct {
+		expected map[string]internal.Position
+	}{
+		{
+			expected: map[string]internal.Position{
+				positionIDs[0]: p,
+			},
+		},
+	}
+	for _, testCase := range testTable {
+		result := repos.GetPositions()
+		if !reflect.DeepEqual(result, testCase.expected) {
+			t.Errorf("Error!\n Expected : %#v;\nResult: %#v\n", testCase.expected, result)
+		}
+	}
+}
+
+func TestGetEmployees(t *testing.T) {
+	initData()
+	p := internal.Position{
+		ID:     createPosID(),
+		Name:   "worker",
+		Salary: decimal.New(500, 0),
+	}
+	repos.AddPosition(&p)
+	e := internal.Employee{
+		ID:         createEmpID(),
+		FirstName:  "Nick",
+		LasName:    "Bobs",
+		PositionID: p.ID,
+	}
+	repos.AddEmployee(&e)
+	testTable := []struct {
+		expected map[string]internal.Employee
+	}{
+		{
+			expected: map[string]internal.Employee{
+				employeeIDs[0]: e,
+			},
+		},
+	}
+	for _, testCase := range testTable {
+		result := repos.GetEmployees()
+		if !reflect.DeepEqual(result, testCase.expected) {
+			t.Errorf("Error!\n Expected : %#v;\nResult: %#v\n", testCase.expected, result)
+		}
+	}
+}
+
+func TestAddEmployee(t *testing.T) {
+	initData()
+	p := internal.Position{
+		ID:     createPosID(),
+		Name:   "worker",
+		Salary: decimal.New(500, 0),
+	}
+	repos.AddPosition(&p)
+	e := internal.Employee{
+		ID:         createEmpID(),
+		FirstName:  "Nick",
+		LasName:    "Bobs",
+		PositionID: p.ID,
+	}
+	repos.AddEmployee(&e)
+	newEmployee := internal.Employee{
+		ID:         createEmpID(),
+		FirstName:  "Bread",
+		LasName:    "Brown",
+		PositionID: p.ID,
+	}
+	testTable := []struct {
+		expected map[string]internal.Employee
+		add      internal.Employee
+	}{
+		{
+			add: e,
+			expected: map[string]internal.Employee{
+				employeeIDs[0]: e,
+			},
+		},
+		{
+			add: newEmployee,
+			expected: map[string]internal.Employee{
+				employeeIDs[0]: e,
+				employeeIDs[1]: newEmployee,
+			},
+		},
```

Comment:
Your test cases should be completely independent between each other.

Reply:
+